packages/connection/src/Manager.php

```diff
@@ -43,6 +193,313 @@ public function initialize( $methods ) {
 		$methods;
 	}
 
+	/**
+	 * Since a lot of hosts use a hammer approach to "protecting" WordPress sites,
+	 * and just blanket block all requests to /xmlrpc.php, or apply other overly-sensitive
+	 * security/firewall policies, we provide our own alternate XML RPC API endpoint
+	 * which is accessible via a different URI. Most of the below is copied directly
+	 * from /xmlrpc.php so that we're replicating it as closely as possible.
+	 */
+	public function alternate_xmlrpc() {
+		// phpcs:disable PHPCompatibility.Variables.RemovedPredefinedGlobalVariables.http_raw_post_dataDeprecatedRemoved
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		global $HTTP_RAW_POST_DATA;
+
+		// Some browser-embedded clients send cookies. We don't want them.
+		$_COOKIE = array();
+
+		// A bug in PHP < 5.2.2 makes $HTTP_RAW_POST_DATA not set by default,
+		// but we can do it ourself.
+		if ( ! isset( $HTTP_RAW_POST_DATA ) ) {
+			$HTTP_RAW_POST_DATA = file_get_contents( 'php://input' );
+		}
+
+		// A fix for mozBlog and other cases where '<?xml' isn't on the very first line.
+		if ( isset( $HTTP_RAW_POST_DATA ) ) {
+			$HTTP_RAW_POST_DATA = trim( $HTTP_RAW_POST_DATA );
+		}
+
+		// phpcs:enable
+
+		include_once ABSPATH . 'wp-admin/includes/admin.php';
+		include_once ABSPATH . WPINC . '/class-IXR.php';
+		include_once ABSPATH . WPINC . '/class-wp-xmlrpc-server.php';
+
+		/**
+		 * Filters the class used for handling XML-RPC requests.
+		 *
+		 * @since 3.1.0
+		 *
+		 * @param string $class The name of the XML-RPC server class.
+		 */
+		$wp_xmlrpc_server_class = apply_filters( 'wp_xmlrpc_server_class', 'wp_xmlrpc_server' );
+		$wp_xmlrpc_server       = new $wp_xmlrpc_server_class();
```

Comment:
This looks a bit fragile. Could we have a whitelist of classes we allow here? I'm concerned anyone could hook a malicious class in the `wp_xmlrpc_server_class` filter that could do bad bad things.

Reply:
Added a `@todo` to address this in 5a4a72b6ac6e2610fb3f47041b762ae0e118ea60. It was like that before, so no rush to fix it now.